Mod2/Lessons/Week2/DataRelationships.md

```diff
@@ -0,0 +1,158 @@
+* Brainstorm database tables that might have a relationship between them. For example, songs and artists or people and favorite food. Don't worry about what type of relationship the tables have or how to build that relationship in SQL!
+<!-- Instructor note, this is a great time to do a slack waterfall to have students share what they came up with for relationships. They will be looking back to these in the lab -->
+
+## Review of Databases and One-To-Many Relationships
+
+Before we jump into the many types of relationships data can have, let's take a step back to review what a database _is_.  At the highest level, a database is a location where we can store information and allow that information to persist in our application.  For today's example, we have a database that is storing song information.  More specifically, we have a `database` that has a songs `table` and that table has columns that store each song's `attributes`; we could visualize this configuration like this:
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_1.png'>
+</p>
+
+In our database, we have a table that looks similar to an excel sheet where each row of this table is one song entry or record in our database. So far, this is a fairly simple database, holding on to only a few attributes.  Without much difficulty, we could update our database to hold additional song attributes by adding additional columns to our songs table.
+
+But, what if we want to keep track of the artist that wrote each song?  We _could_ add a column to our database called `artist` and store the artist directly on a song, like this:
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_2.png'>
+</p>
+
+Seems like it works, right? Well... not exactly.  What if we wanted to also store information about specific artists, like their hometown or years active.  If we imagine what that might look like, we would see more columns on our `songs` table and many of those columns would be storing the exact same information - every Prince song would be storing 'Minneapolis' and '1975-2016':
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_3.png'>
+</p>
+
+This is very repetitive!
+
+## Normalizing our Database
+
+In order to cut down on redundant data on a specific table in our database, we can rely on **normalization**.  Normalization is a process by which we break down tables in our database so that each table relates only to one resource.  In the example we are working with, when we look at our songs table, we are storing information about songs and about artists. Really, we are trying to store information about two resources on one table - let's fix that.
+
+In your pairs, draw how you would normalize this database, so that songs and artists each have their own table. I highly recommend you look back at the Intro to Databases lesson to see how we normalized customers and orders!
```

Comment:
Ahhh normalization. How far down this rabbit hole will we go, or is this as far as we want? First Normal Form (1NF), Second, Third, Boyce-Codd, etc. Sorry, I had an undergrad flashback.

Reply:
I took most of this content from the backend lessons, I'm on the fence if normalization is a term students need to know. But because we already introduced this concept in Intro to Databases, I think it's good to introduce the additional term here so it doesn't feel like only review.